services/app-web/src/otelHelpers/tracingHelper.ts

```diff
@@ -0,0 +1,61 @@
+import opentelemetry, { Span } from '@opentelemetry/api'
+import { User as UserType } from '../gen/gqlClient'
+
+const serviceNameOTEL = 'app-web-' + process.env.REACT_APP_STAGE_NAME
+
+function getTracer() {
+    const tracer = opentelemetry.trace.getTracer(serviceNameOTEL)
+    return tracer
+}
+
+// Add a span to existing trace or if none, start and end and new span
+// if a callback is passed in, end span after that async action, this could allow the trace to also collect how long the action took
+// based on https://blog.devgenius.io/measuring-react-performance-with-opentelemetry-and-honeycomb-2b20a7920335
+async function recordSpan(
+    name: string,
+    cb?: () => Promise<void>,
+    parentSpan?: Span
+): Promise<void> {
+    const tracer = getTracer()
+    let span: Span
+    if (parentSpan) {
+        const otelContext = opentelemetry.trace.setSpan(
+            opentelemetry.context.active(),
+            parentSpan
+        )
+        span = tracer.startSpan(name, undefined, otelContext)
+    } else {
+        span = tracer.startSpan(name)
+    }
+
+    if (cb) {
+        try {
+            await cb()
+        } catch (err) {
+            span.recordException(err)
+        }
+    }
+
+    span.end()
+
+    return
+}
+
+function recordJSException(error: string | Error): void {
```

Comment:
I tried to figure out if there was a fast way to pass in the ReactComponent name automatically to this function but I didn't see one. I think it requires adding `displayName` manually to each component which seemed like a bigger change.

I noted that  one of the [React auto-instrumentation tools](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/examples/react-load/react) for OTEL (which we aren't using because of the class components) uses a class components based way to add the name but there is an open ticket about creating a hook for use with functional components.

Reply:
Yeah, when I first went about putting this together I was hoping we could use that lib, but that issue asking for functional component support has been open for two years 😭